### PR TITLE
stats: add periodic task to generate reports 

### PR DESCRIPTION
**Pull request by jma**
* Renames private method.
* Adds periodic task for monthly and yearly.
* Adds cli to generate synchronous and asynchronous reports.
* Enables all tasks but ebook during the setup.

### CODE REVIEW

**Review by lauren-d — APPROVED** [review]

File: rero_ils/modules/stats/api/report.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
 from .indicators import NumberOfActivePatronsCfg, NumberOfCirculationCfg, \
     NumberOfDeletedItemsCfg, NumberOfDocumentsCfg, NumberOfILLRequests, \
     NumberOfItemsCfg, NumberOfPatronsCfg, NumberOfSerialHoldingsCfg
+from ..api.api import Stat
```
Comment (zannkukai) [inline rero_ils/modules/stats/api/report.py:33]: `..api.api`😅 This is a great mental gymnastic 🤯

**Review by zannkukai — APPROVED** [review]
This PR contains 2 commit, I only reviewed the 'periodic task' commit

File: rero_ils/config.py [hunk 2]
```diff
@@ -420,6 +420,22 @@ def _(x):
         'schedule': crontab(minute=30, hour=1, day_of_month='1'),  # First day of the month at 01:30 UTC,
         'enabled': False
     },
+    'collect-stats-report-month': {
+        'task': ('rero_ils.modules.stats.tasks.collect_stats_reports'),
+        'schedule': crontab(minute=00, hour=1, day_of_month='1'),  # First day of the month at 01:30 UTC,
```
Comment (rerowep) [inline rero_ils/config.py:425]: Is it working with `00`?

File: rero_ils/modules/stats/cli.py [hunk 3]
```diff
@@ -227,17 +227,23 @@ def collect(pid):
...
-                'from': range['gte'],
-                'to': range['lte']
-            }
-
-        Stat.create(data, dbcommit=True, reindex=True)
+        values = stat_report.collect()
+        stat_report.create_stat(values)
+
+
+@report.command('collect-all')
```
Comment (rerowep) [inline rero_ils/modules/stats/cli.py:234]: Could be written `@report.command()`

File: rero_ils/modules/stats/cli.py [hunk 4]
```diff
@@ -227,17 +227,23 @@ def collect(pid):
...
+    """Extract the stats report values and store it.
+
+    :param pid: pid value of the configuration to use.
+    """
+    from .tasks import collect_stats_reports
+    if delayed:
+        collect_stats_reports.delay(frequency)
+    else:
+        res = collect_stats_reports(frequency)
+        click.secho(f'Generated {len(res)} reports.')
```
Comment (rerowep) [inline rero_ils/modules/stats/cli.py:249]: Use `click.echo`. `click.secho` is used if color is needed.

File: rero_ils/modules/stats/cli.py [hunk 5]
```diff
@@ -227,17 +227,23 @@ def collect(pid):
...
+              help='Run indexing in background.')
+@with_appcontext
+def collect_all(frequency, delayed):
+    """Extract the stats report values and store it.
+
+    :param pid: pid value of the configuration to use.
+    """
+    from .tasks import collect_stats_reports
+    if delayed:
+        collect_stats_reports.delay(frequency)
```
Comment (rerowep) [inline rero_ils/modules/stats/cli.py:246]: Maybe print the task id like this:
```python
res =collect_stats_reports.delay(frequency)
click.secho(f'Generated reports delayed: {res}')
```

File: rero_ils/modules/stats/tasks.py [hunk 6]
```diff
@@ -52,3 +56,28 @@ def collect_stats_librarian():
             dbcommit=True, reindex=True)
         return f'New statistics of type {stat["type"]} has\
             been created with a pid of: {stat.pid}'
+
+
+@shared_task()
+def collect_stats_reports(frequency='month'):
+    """Collect and store the montly statistics for librarian."""
```
Comment (rerowep) [inline rero_ils/modules/stats/tasks.py:63]: Maybe test if `frequency` is `month` or `year`?

**Review by rerowep — CHANGES_REQUESTED** [review]

**Review by rerowep — APPROVED** [review]

**Pull request merged**